test/controllers/status-bar-tile-controller.test.js

```diff
@@ -646,6 +646,50 @@ describe('StatusBarTileController', function() {
     });
   });
 
+  describe('when the local branch is named differently from the remote branch it\'s tracking', function() {
+    let repository, wrapper;
+    beforeEach(async function() {
+      const {localRepoPath} = await setUpLocalAndRemoteRepositories();
+      repository = await buildRepository(localRepoPath);
+      wrapper = await mountAndLoad(buildApp({repository}));
+      await repository.git.exec(['checkout', '-b', 'another-name', '--track', 'origin/master']);
+      repository.refresh();
+    });
+
+    it('fetches with no git error', async function() {
+      sinon.spy(repository, 'fetch');
+      await wrapper
+        .instance()
+        .fetch(await wrapper.instance().fetchData(repository))();
+      assert.isTrue(repository.fetch.called);
+    });
+
+    it('pulls from the correct branch', async function() {
+      const prePullSHA = await repository.git.exec(['rev-parse', 'HEAD']);
+      await repository.git.exec(['reset', '--hard', 'HEAD~2']);
+      sinon.spy(repository, 'pull');
+      await wrapper
+        .instance()
+        .pull(await wrapper.instance().fetchData(repository))();
+      const postPullSHA = await repository.git.exec(['rev-parse', 'HEAD']);
+      assert.isTrue(repository.pull.called);
+      assert.equal(prePullSHA, postPullSHA);
+    });
+
+    it('pushes to the correct branch', async function() {
+      await repository.git.commit('new local commit', {allowEmpty: true});
+      const localSHA = await repository.git.exec(['rev-parse', 'another-name']);
+      sinon.spy(repository, 'push');
+      await wrapper
+        .instance()
+        .push(await wrapper.instance().fetchData(repository))();
+      const remoteSHA = await repository.git.exec(['rev-parse', 'origin/master']);
+      assert.isTrue(repository.push.called);
```

Comment:
same nit as above - might be good to specify which args `push` should be called with.